cmd/notation/plugin.go

```diff
@@ -62,3 +108,169 @@ func listPlugins(command *cobra.Command) error {
+	_, err = os.Stat(pluginDir + "/" + pluginBinary)
+
+	// copy plugin, if not exist
+	if os.IsNotExist(err) {
+		copyPlugin(pluginPath, pluginDir+"/"+pluginBinary)
+	}
+
+	// overwrite plugin, if force flag is set
+	if err == nil && force {
+		fmt.Printf("Overwriting plugin %s in directory %s\n", pluginBinary, pluginDir)
+		copyPlugin(pluginPath, pluginDir+"/"+pluginBinary)
+	}
+
+	// if plugin exists and force flag is not set, get plugin metadata
+	if err == nil && !force {
+		cmd := exec.Command(pluginDir+"/"+pluginBinary, "get-plugin-metadata")
+
+		output, err := cmd.Output()
+		if err != nil {
+			return err
+		}
+
+		var currentPlugin map[string]interface{}
+		err = json.Unmarshal([]byte(output), &currentPlugin)
+		if err != nil {
+			return err
+		}
+
+		// check if new plugin version is greater than current plugin version
+		currentPluginVersion := currentPlugin["version"].(string)
+		currentVersion, err := semver.NewVersion(currentPluginVersion)
+		if err != nil {
+			return err
+		}
+
+		// copy plugin, if new plugin version is greater than current plugin version
+		if newSemVersion.GreaterThan(currentVersion) {
+			var confirm string
+
+			fmt.Printf("Detected new version %s. Current version is %s.\nDo you want to overwrite the plugin %s? [y/N]: ", newSemVersion.String(), currentVersion.String(), pluginName)
+			fmt.Scanln(&confirm)
+
+			if strings.ToLower(confirm) != "y" {
+				fmt.Println("Operation cancelled.")
+				return nil
+			}
+
+			fmt.Printf("Copying plugin %s to directory %s...\n", pluginName, pluginDir)
+			copyPlugin(pluginPath, pluginDir+"/"+pluginBinary)
+		}
+
+		// do not copy plugin, if new plugin version is less than or equal to current plugin version
+		if newSemVersion.LessThan(currentVersion) || newSemVersion.Equal(currentVersion) {
+			fmt.Println("Current version is greater than or equal to new version. Skipping plugin installation.\nUse --force flag to overwrite the plugin.")
+		}
+	}
+
+	return nil
+}
+
+func removePlugin(command *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return errors.New("missing plugin name")
+	}
+
+	pluginName := args[0]
+
+	// get plugin directory
+	pluginDir, err := dir.PluginFS().SysPath(pluginName)
+	if err != nil {
+		return err
+	}
+
+	// Check if plugin directory exists
+	_, err = os.Stat(pluginDir)
+	if os.IsNotExist(err) {
+		return errors.New("plugin does not exist")
+	}
+
+	// remove plugin directory
+	err = os.RemoveAll(pluginDir)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func copyPlugin(src, dest string) error {
+	sourceFile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+
+	sourceFileInfo, err := sourceFile.Stat()
+	if err != nil {
+		return err
+	}
+	fileMode := sourceFileInfo.Mode()
+
+	destFile, err := os.OpenFile(dest, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fileMode)
+	if err != nil {
+		return err
+	}
+	defer destFile.Close()
+
+	_, err = io.Copy(destFile, sourceFile)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
Can we try to use existing function `CopyToDir()` here?https://github.com/notaryproject/notation/blob/0ec3b3dd7166e680056b14b23847a03976edcfea/internal/osutil/file.go#L43

Reply:
Yes, we need to reuse code.

Reply:
Hey @patrickzheng200, thanks for letting me know about that internal helper. I [replaced](https://github.com/notaryproject/notation/pull/617/commits/3eb146c202f2e6a75d4184f5c79d43cd42bc80e5)the logic in the install command to use the CopyToDir func.